src/acp/matcher.test.ts

```diff
@@ -141,14 +140,20 @@ const MOCK_CLIENT_WEBID_1 = DataFactory.namedNode(
 const MOCK_CLIENT_WEBID_2 = DataFactory.namedNode(
   "https://your.app/registration#it"
 );
+const MOCK_CLIENT_WEBID_3 = DataFactory.namedNode("test_string_client_id");
+const MOCK_CLIENT_WEBID_4 = DataFactory.namedNode("other_string_client_id");
 
 const addAllObjects = (
   thing: ThingPersisted,
   predicate: NamedNode,
   objects: Url[]
 ): ThingPersisted => {
   return objects.reduce((thingAcc, object) => {
-    return addUrl(thingAcc, predicate, object);
+    try {
+      return addUrl(thingAcc, predicate, object);
+    } catch (e) {
+      return addStringNoLocale(thingAcc, predicate, object.value);
+    }
```

Comment:
Wouldn't it be cleaner to use `internal_isValidUrl` ? Even if this works, I find it a bit surprising ^^.

Reply:
Oh absolutely ^^,

Reply:
To be fair, that's *exactly* how `internal_isValidUrl` is implemented, but still :p